### PR TITLE
41 Backend Class Diagram

### PR DESCRIPTION
**Pull request by pontuslaestadius**
Does not work with lobbies.

Identify in the backend if you have received a class diagram.
Link the class diagram with the sequence diagram.
When a message is sent, NOTIFY the frontend to highlight the class.
Store the class information in the nodes
Print on the client when you have sucessfully linked a diagram.

### CODE REVIEW

File: Application/src/controller/network/Decode.java [hunk 1]
```diff
@@ -32,6 +33,16 @@ void execute() {
         // If no string has been allocated, abort.
         if (rawStringToDecode == null) return;
 
+        // TODO remove when 39 is done.
```
Comment (timjon) [inline Application/src/controller/network/Decode.java:35]: Do we really want test code?

File: Application/src/model/Parser.java [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 import model.classDiagram.*;
 import model.sequenceDiagramParser.*;
+
+import view.DiagramView;
+
```
Comment (timjon) [inline Application/src/model/Parser.java]: No version update

File: Application/src/model/Parser.java [hunk 3]
```diff
@@ -62,20 +65,25 @@ public void parseClassDiagram(String inputJSON) {
...
+            string.append("class_diagram,");
+            string.append(DiagramView.getDiagramViewInView().getTab().getId());
+            string.append(",");
             string.append(UniqueCounter.getString());
             string.append(",[");
 
             // Add the classes.
             for (Classes s: cd.classes) {
-                string.append("[");
+                string.append("['");
```
Comment (timjon) [inline Application/src/model/Parser.java:117]: Why did you change this?

File: Application/src/model/classDiagram/Relationships.java [hunk 4]
```diff
@@ -14,6 +14,6 @@
      * @return a formatted string.
      */
     public String format() {
-        return "[" + type + "," + superclass + "," + subclass + "]";
+        return "['" + type + "','" + superclass + "','" + subclass + "']";
```
Comment (timjon) [inline Application/src/model/classDiagram/Relationships.java:17]: Why is this changed?

File: Application/src/model/classDiagram/Relationships.java [hunk 5]
```diff
@@ -14,6 +14,6 @@
      * @return a formatted string.
      */
     public String format() {
-        return "[" + type + "," + superclass + "," + subclass + "]";
```
Comment (timjon) [inline Application/src/model/classDiagram/Relationships.java:17]: No version update

File: Application/src/view/visuals/Draw.java [hunk 6]
```diff
@@ -371,4 +371,31 @@ public Message getMessage(int index) throws ArrayIndexOutOfBoundsException {
             throw new ArrayIndexOutOfBoundsException("Index: " + index + "Size: " + this.messages.size());
         return this.messages.get(index);
     }
+
+    /**
+     * Given a name will return the class that created it.
+     * @param name to convert.
+     * @return the class name that made it.
+     */
+    public String convertInstantiatedClassToClassIdentifier(String name) throws NullPointerException {
```
Comment (timjon) [inline Application/src/view/visuals/Draw.java]: Theres a lot of unnecessary empty lines in this method

File: Application/src/view/visuals/component/Message.java [hunk 7]
```diff
@@ -260,7 +260,7 @@ public void renderDefault(GraphicsContext gc) {
         // int y2 = this.node2.getY(); //Not used atm.
 
         y1 += offset; // Sets an offset from the previous message.
-
+
```
Comment (timjon) [inline Application/src/view/visuals/component/Message.java:263]: No version update

File: Server/src/compile.sh [hunk 8]
```diff
@@ -1,14 +1,18 @@
 # Author: Pontus Laestadius
-# Version: 1.0
+# Version: 1.2
```
Comment (timjon) [inline Server/src/compile.sh]: Why did the version change from 1.0 to 1.2

File: Server/src/compile.sh [hunk 9]
```diff
@@ -1,14 +1,18 @@
...
+
+# Intro message.
+echo "-----------ERLANG COMPILATION SCRIPT VERSION 1.2-----------"
+FINISHED="-------------------------FINISHED!-------------------------"
```
Comment (timjon) [inline Server/src/compile.sh]: The intro message says finished?

File: Server/src/diagramcoordinator.erl [hunk 10]
```diff
@@ -2,89 +2,183 @@
...
-	  loop(Coordinator, Did ,Pids,[],Message_number, PrevList);
+  
+  % Add a class diagram
+  {class_diagram, Classid, Classes, Relations, Coordinator} ->
+		Coordinator ! {class_diagram, Classid, Did, {Classes, Relations}},
+	  Coordinator !  {Did, print_information, ["Linked Class diagram: " ++ atom_to_list(name_classes(Pids, Classes))]},
+  	loop(Coordinator, Did, Pids, [], Message_number, PrevList, {Classid, {Classes, Relations}});
+  
+  
+  {next_message, Coordinator} ->
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl]: Sorry im being stupid but what does this case do?

File: Server/src/diagramcoordinator.erl [hunk 11]
```diff
@@ -2,89 +2,183 @@
...
+loop(Coordinator, Did, Pids, [], Message_number, PrevList, ClassDiagram) ->
   receive
-    {next_message, Coordinator} -> 
-      Coordinator ! {simulation_done, Did, Message_number},
-	  loop(Coordinator, Did ,Pids,[],Message_number, PrevList);
+  
+  % Add a class diagram
+  {class_diagram, Classid, Classes, Relations, Coordinator} ->
+		Coordinator ! {class_diagram, Classid, Did, {Classes, Relations}},
+	  Coordinator !  {Did, print_information, ["Linked Class diagram: " ++ atom_to_list(name_classes(Pids, Classes))]},
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl]: Why does it send 2 messages to the same thing?

File: Server/src/diagramcoordinator.erl [hunk 12]
```diff
@@ -2,89 +2,183 @@
...
     {next_message, Coordinator} -> 
       {From, To, Message} = L,
-	  send_message(find_pid(Pids, From), From, To, Message, find_pid(Pids, To), Message_number, Coordinator, Did), 
+	  send_message(find_pid(Pids, From), From, To, Message, find_pid(Pids, To), Message_number, Coordinator, Did, get_class_diagram_id(ClassDiagram)), 
       receive
-	    {message_done, From, To, Message, Message_number} ->
-		  Coordinator ! {message_sent, Did, From, To, Message, Message_number},
-		  %Sends info to the Coordinator that a message has been received by a node. To be printed in the executionlog
-		  Coordinator !  {Did, print_information, ["Node " ++ atom_to_list(To) ++ " received a message from " ++ atom_to_list(From)]}
+	    	{message_done, From, To, Message, Message_number} ->
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl:106]: Indentation change or just git?

File: Server/src/diagramcoordinator.erl [hunk 13]
```diff
@@ -2,89 +2,183 @@
...
+		  	Coordinator ! {message_sent, Did, From, To, Message, Message_number},
+		  	%Sends info to the Coordinator that a message has been received by a node. To be printed client-side.
+		  	Coordinator !  {Did, print_information, ["Node " ++ atom_to_list(To) ++ " received a message from " ++ atom_to_list(From)]}
 	  end,
-	  loop(Coordinator, Did, Pids, Ls, Message_number + 1, [L|PrevList]);
+	  loop(Coordinator, Did, Pids, Ls, Message_number + 1, [L|PrevList], ClassDiagram);
+
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl]: Is this empty line really needed?

File: Server/src/diagramcoordinator.erl [hunk 14]
```diff
@@ -2,89 +2,183 @@
...
+	  
 	  
 	{previous_message, Coordinator} ->
 	  [Prev_H| Prev_T] = PrevList,
 	  List = [L|Ls],
 	  Coordinator ! {previous_confirmation, Did, ["Previous message"]},
-	  loop(Coordinator, Did, Pids, [Prev_H| List], Message_number - 1, Prev_T)
+	  loop(Coordinator, Did, Pids, [Prev_H| List], Message_number - 1, Prev_T, ClassDiagram)
   end.
+
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl]: Another wierd empty line

File: Server/src/diagramcoordinator.erl [hunk 15]
```diff
@@ -2,89 +2,183 @@
...
   
-%checks if a class has a process and spawns it if it doesnt exist
+%checks if a class has a process, none if it does not exist.
+find_pid([], _) -> none;
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl:123]: Have you checked the other places this is used so it doesnt break something else?

File: Server/src/diagramcoordinator.erl [hunk 16]
```diff
@@ -2,89 +2,183 @@
...
 
+
+% Gets the name of a specific node.
+getName(NodePid) ->
+	NodePid ! {self(), getName},
+	receive 
+		{getName, Name} -> Name
+		
+	after
+		1000 ->
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl]: why this number?

File: Server/src/diagramcoordinator.erl [hunk 17]
```diff
@@ -2,89 +2,183 @@
...
 find_pid([_|Ls], Name)                      -> find_pid(Ls, Name). 
 
+
+% Gets the name of a specific node.
+getName(NodePid) ->
+	NodePid ! {self(), getName},
+	receive 
+		{getName, Name} -> Name
+		
+	after
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl]: Do you really need a timeout?

File: Server/src/diagramcoordinator.erl [hunk 18]
```diff
@@ -2,89 +2,183 @@
...
+
+% Returns the wrapped Class diagram id or none.
+get_class_diagram_id(none) -> none;
+get_class_diagram_id({Id, {_, _}}) -> Id.
+
+
 %spawns nodes for every class in the list
 spawn_nodes(List, Did, Coordinator) ->[spawn_node(Class, Did, Coordinator) || Class <- List].
 
+
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl:135]: wow

File: Server/src/diagramcoordinator.erl [hunk 19]
```diff
@@ -2,89 +2,183 @@
...
 %spawns a node and returns a tuple with the pid and the class name
 spawn_node(Class_name, Did, Coordinator) -> 
   Self = self(),
   %Sends info to the Coordinator, that a node has been spawned. To be printed in the executionlog
   Coordinator ! {Did, print_information, ["Spawned node " ++ atom_to_list(Class_name)]},
-  {node:init(Self), Class_name}.
+  {node:init(Self, Class_name), Class_name}.
+
+
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl:143]: wow

File: Server/src/diagramcoordinator.erl [hunk 20]
```diff
@@ -2,89 +2,183 @@
...
+% Iterates over the Pids and classes and propogates the data to the Pid.
+name_classes([{H1, _}], [H2]) -> H1 ! {set, H2}, ok;
+name_classes([{H1, _}|T1], [H2|T2]) -> 
+	H1 ! {set, H2},
+	name_classes(T1, T2);
+name_classes(_,_) -> err.
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl]: what does this do

File: Server/src/diagramcoordinator.erl [hunk 21]
```diff
@@ -2,89 +2,183 @@
...
 spawn_node(Class_name, Did, Coordinator) -> 
   Self = self(),
   %Sends info to the Coordinator, that a node has been spawned. To be printed in the executionlog
   Coordinator ! {Did, print_information, ["Spawned node " ++ atom_to_list(Class_name)]},
-  {node:init(Self), Class_name}.
+  {node:init(Self, Class_name), Class_name}.
+
+
+% Iterates over the Pids and classes and propogates the data to the Pid.
+name_classes([{H1, _}], [H2]) -> H1 ! {set, H2}, ok;
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl]: The variable names seems kinda confusing here

File: Server/src/diagramcoordinator.erl [hunk 22]
```diff
@@ -2,89 +2,183 @@
...
+name_classes([{H1, _}|T1], [H2|T2]) -> 
+	H1 ! {set, H2},
+	name_classes(T1, T2);
+name_classes(_,_) -> err.
+	
+	
+% Notifies the client if there is a class diagram, to highlight a specific class.
+notify_class_diagram(Coordinator, Did, ClassDiagramId, Pid) ->
```
Comment (timjon) [inline Server/src/diagramcoordinator.erl]: A lot of empty lines here

File: Server/src/lobbycoordinator.erl [hunk 23]
```diff
@@ -100,4 +100,4 @@ get_lobby_ID([_|Ls]) -> get_lobby_ID(Ls).
 %Helper function that takes all numbers until another char appears
 get_ID([]) -> [];
 get_ID([L|Ls]) when (L >= $0) and (L =< $9) -> [L| get_ID(Ls)];
-get_ID(_) -> [].
\ No newline at end of file
+get_ID(_) -> [].
```
Comment (timjon) [inline Server/src/lobbycoordinator.erl:104]: Version update?

**Review by timjon — CHANGES_REQUESTED** [review]
I didnt have time to go through everything sadly

File: Application/src/controller/network/Decode.java (hunk 1 above)
Comment (Sebastian-Fransson) [reply Application/src/controller/network/Decode.java:35]: +1

File: Server/src/node.erl [hunk 24]
```diff
@@ -1,24 +1,41 @@
...
+%%Version: 1.4
 
 %Initialized the process for the node
-init(Coordinator) ->
-  spawn(fun () ->  loop(Coordinator) end).
+init(Coordinator, Name) ->
+  spawn(fun () ->  loop(Coordinator, Name, none, [], []) end).
 
-loop(Coordinator) ->
+loop(Coordinator, Name, Class, Fields, Relations) ->
```
Comment (timjon) [inline Server/src/node.erl]: Do you really need the fields and relations here? you are not using them

File: Server/src/usercoordinator.erl [hunk 25]
```diff
@@ -53,6 +54,12 @@ loop(Socket, Diagrams) ->
 		Format_result = io_lib:format("~p", [{Did, print_information, Msg}]),
 		%Sends the result to client
 		gen_tcp:send(Socket, [Format_result ++ "~"]),
+		loop(Socket, Diagrams);
+		
+		% Highlight a specific class.
+	{class_diagram, Did, SeId, highlight, Name} ->
```
Comment (timjon) [inline Server/src/usercoordinator.erl]: What is the SeId variable for?

File: Server/src/diagramcoordinator.erl [hunk 26]
```diff
@@ -2,89 +2,183 @@
...
+getName(NodePid) ->
+	NodePid ! {self(), getName},
+	receive 
+		{getName, Name} -> Name
+		
+	after
+		1000 ->
+		  err
+	end.
+
```
Comment (IsabelleTornqvist) [inline Server/src/diagramcoordinator.erl]: Agree with Tim: looots of weird, unneccessary empty lines?

File: Server/src/diagramcoordinator.erl [hunk 27]
```diff
@@ -2,89 +2,183 @@
...
+	
+		% Matches if it has a name or not.
+		case Name of
+	
+			% None has been provided.
+			none  -> none;
+		
+			% Node did not responde.
+			err -> err;
+
```
Comment (IsabelleTornqvist) [inline Server/src/diagramcoordinator.erl]: Why

File: Application/src/controller/network/Decode.java (hunk 1 above)
Comment (pontuslaestadius) [reply Application/src/controller/network/Decode.java:35]: No this is for Kosara fuck off

File: Application/src/model/Parser.java (hunk 3 above)
Comment (pontuslaestadius) [reply Application/src/model/Parser.java:117]: Because a developer requested a change which was require to be Erlang compatible.

File: Application/src/model/classDiagram/Relationships.java (hunk 4 above)
Comment (pontuslaestadius) [reply Application/src/model/classDiagram/Relationships.java:17]: Atoms need 'atoms'

File: Application/src/view/visuals/component/Message.java (hunk 7 above)
Comment (pontuslaestadius) [reply Application/src/view/visuals/component/Message.java:263]: I hope you have a bright future ahead of you!

File: Server/src/compile.sh (hunk 9 above)
Comment (pontuslaestadius) [reply Server/src/compile.sh]: No you kind person learn shell scripting. I am just making a variable called FINISHED

File: Server/src/diagramcoordinator.erl (hunk 12 above)
Comment (pontuslaestadius) [reply Server/src/diagramcoordinator.erl:106]: Indention perfect

File: Server/src/diagramcoordinator.erl (hunk 13 above)
Comment (pontuslaestadius) [reply Server/src/diagramcoordinator.erl]: So LOC * division of atan() results in yes.

File: Server/src/diagramcoordinator.erl (hunk 14 above)
Comment (pontuslaestadius) [reply Server/src/diagramcoordinator.erl]: Git recommends ending files with empty lines.

File: Server/src/diagramcoordinator.erl (hunk 16 above)
Comment (pontuslaestadius) [reply Server/src/diagramcoordinator.erl]: One second seems reasonable.

File: Application/src/model/classDiagram/Relationships.java [hunk 28]
```diff
@@ -3,7 +3,7 @@
 /**
  * Stores fetchable data used for the GSON parser.
  * @author Pontus Laestadius
- * @version 1.0
+ * @version 3.1
```
Comment (timjon) [inline Application/src/model/classDiagram/Relationships.java]: 1.0 to 3.1?

**Review by timjon — CHANGES_REQUESTED** [review]

File: Application/src/model/Parser.java (hunk 3 above)
Comment (Sebastian-Fransson) [reply Application/src/model/Parser.java:117]: Who is idiot? specify

File: Application/src/model/classDiagram/Relationships.java (hunk 28 above)
Comment (Sebastian-Fransson) [reply Application/src/model/classDiagram/Relationships.java]: +1

File: Server/src/compile.sh (hunk 8 above)
Comment (Sebastian-Fransson) [reply Server/src/compile.sh]: +1

File: Server/src/diagramcoordinator.erl [hunk 29]
```diff
@@ -2,89 +2,192 @@
...
-loop(Coordinator, Did, Pids, [], Message_number, PrevList) ->
+loop(Coordinator, Did, Pids, [], Message_number, PrevList, ClassDiagram) ->
   receive
-    {next_message, Coordinator} -> 
-      Coordinator ! {simulation_done, Did, Message_number},
-	  loop(Coordinator, Did ,Pids,[],Message_number, PrevList);
+  
+  % Add a class diagram
+  {class_diagram, Classid, Classes, Relations, Coordinator} ->
+		Coordinator ! {class_diagram, Classid, Did, {Classes, Relations}},
```
Comment (Sebastian-Fransson) [inline Server/src/diagramcoordinator.erl:26]: Sends twice?

File: Server/src/diagramcoordinator.erl [hunk 30]
```diff
@@ -2,89 +2,192 @@
...
   end;
  
  %When there are no previous messages this case is used.
-  loop(Coordinator, Did, Pids, [L|Ls], Message_number, []) ->
+  loop(Coordinator, Did, Pids, [L|Ls], Message_number, [], ClassDiagram) ->
     receive
+    
+    % Add a class diagram
+  {class_diagram, Classid, Classes, Relations, Coordinator} ->
+		Coordinator ! {class_diagram, Classid, Did, {Classes, Relations}},
```
Comment (Sebastian-Fransson) [inline Server/src/diagramcoordinator.erl:55]: Sends twice? comment on why?

File: Server/src/diagramcoordinator.erl [hunk 31]
```diff
@@ -2,89 +2,192 @@
...
   end;
   
 %This loop runs until the list is empty (when there are no more messages)
-loop(Coordinator, Did, Pids, [L|Ls], Message_number, PrevList) -> 
+loop(Coordinator, Did, Pids, [L|Ls], Message_number, PrevList, ClassDiagram) -> 
   receive
+
+	% Add a class diagram
+  {class_diagram, Classid, Classes, Relations, Coordinator} ->
+		Coordinator ! {class_diagram, Classid, Did, {Classes, Relations}},
```
Comment (Sebastian-Fransson) [inline Server/src/diagramcoordinator.erl:91]: Same here

Comment (Sebastian-Fransson) [reply Server/src/diagramcoordinator.erl:91]: Comment why

File: Server/src/node.erl (hunk 24 above)
Comment (Sebastian-Fransson) [reply Server/src/node.erl]: Could be compacted?

File: Server/src/usercoordinator.erl [hunk 32]
```diff
@@ -53,6 +54,12 @@ loop(Socket, Diagrams) ->
 		Format_result = io_lib:format("~p", [{Did, print_information, Msg}]),
 		%Sends the result to client
 		gen_tcp:send(Socket, [Format_result ++ "~"]),
+		loop(Socket, Diagrams);
+		
+		% Highlight a specific class.
+	{class_diagram, Did, SequenceDiagramId, highlight, Name} ->
+	  Format_result = io_lib:format("~p", [{class_diagram, Did, SequenceDiagramId, highlight_class_diagram, Name}]) ++ "~",
+		gen_tcp:send(Socket, Format_result),
```
Comment (Sebastian-Fransson) [inline Server/src/usercoordinator.erl:62]: what happens here? (reader iq 0.5)

**Review by Sebastian-Fransson — CHANGES_REQUESTED** [review]

File: Application/src/model/classDiagram/Relationships.java (hunk 28 above)
Comment (IsabelleTornqvist) [reply Application/src/model/classDiagram/Relationships.java]: +1

File: Application/src/view/visuals/component/Message.java [hunk 33]
```diff
@@ -10,7 +10,7 @@
  * Class for creating the messages to pass between "classes".
  * @author Sebastian Fransson
  * Collaborator Rashad Kamsheh, Isabelle Törnqvist, Pontus Laestadius, Tim Jonasson
- * @version 4.3
+ * @version 4.3.01
```
Comment (IsabelleTornqvist) [inline Application/src/view/visuals/component/Message.java]: wat

File: Server/src/compile.sh (hunk 9 above)
Comment (IsabelleTornqvist) [reply Server/src/compile.sh]: Popo, respecc thy coworkers

File: Application/src/model/Parser.java (hunk 2 above)
Comment (kosara-g) [reply Application/src/model/Parser.java]: +1

File: Application/src/model/classDiagram/Relationships.java [hunk 34]
```diff
@@ -1,19 +1,19 @@
 package model.classDiagram;
 
 /**
- * Stores fetchable data used for the GSON parser.
+ * Stores fetch-able data used for the GSON parser.
```
Comment (kosara-g) [inline Application/src/model/classDiagram/Relationships.java:4]: no, it's definitely "fetchable"

File: Application/src/model/classDiagram/Relationships.java (hunk 5 above)
Comment (kosara-g) [reply Application/src/model/classDiagram/Relationships.java:17]: there is one now

File: Application/src/view/visuals/Draw.java [hunk 35]
```diff
@@ -274,14 +274,6 @@ private void renderMessage() {
...
     }
 
-    /**
-     * checks a canvas properties to validate if it is in view or not.
-     * @param canvas to check
-     */
-    private boolean inView(Canvas canvas) {
-        return canvas.getWidth() == 0 && canvas.getHeight() == 0;
-    }
-
```
Comment (kosara-g) [inline Application/src/view/visuals/Draw.java:285]: why was this removed?

File: Application/src/view/visuals/component/Message.java (hunk 33 above)
Comment (kosara-g) [reply Application/src/view/visuals/component/Message.java]: +1

File: Server/src/compile.sh (hunk 8 above)
Comment (kosara-g) [reply Server/src/compile.sh]: +1

File: Server/src/diagramcoordinator.erl (hunk 30 above)
Comment (kosara-g) [reply Server/src/diagramcoordinator.erl:55]: +1

File: Server/src/diagramcoordinator.erl [hunk 36]
```diff
@@ -1,90 +1,208 @@
...
-	  send_message(find_pid(Pids, From), From, To, Message, find_pid(Pids, To), Message_number, Coordinator, Did), 
-      receive
-	    {message_done, From, To, Message, Message_number} ->
-		  Coordinator ! {message_sent, Did, From, To, Message, Message_number},
-		  %Sends info to the Coordinator that a message has been received by a node. To be printed in the executionlog
-		  Coordinator !  {Did, print_information, ["Node " ++ atom_to_list(To) ++ " received a message from " ++ atom_to_list(From)]}
+      
+	  	send_message(find_pid(Pids, From), From, To, Message, find_pid(Pids, To), Message_number, Coordinator, Did, get_class_diagram_id(ClassDiagram)), 
+      	receive
+
```
Comment (kosara-g) [inline Server/src/diagramcoordinator.erl:70]: why are the cases below indented so far in?

File: Server/src/diagramcoordinator.erl (hunk 31 above)
Comment (kosara-g) [reply Server/src/diagramcoordinator.erl:91]: +1

**Review by kosara-g — CHANGES_REQUESTED** [review]
Please refer to the comments.

File: Application/src/model/Parser.java (hunk 3 above)
Comment (ghost) [reply Application/src/model/Parser.java:117]: @pontuslaestadius how about String.toLowerCase();

**Review by ghost — COMMENTED** [review]
commit messages are burning

File: Application/src/view/visuals/Draw.java (hunk 35 above)
Comment (pontuslaestadius) [reply Application/src/view/visuals/Draw.java:285]: not used

File: Server/src/diagramcoordinator.erl (hunk 36 above)
Comment (pontuslaestadius) [reply Server/src/diagramcoordinator.erl:70]: open in an IDE

File: Application/src/model/classDiagram/Relationships.java (hunk 34 above)
Comment (pontuslaestadius) [reply Application/src/model/classDiagram/Relationships.java:4]: Intellij didn't like it and I don't like yellow errors.

**Review by timjon — APPROVED** [review]

**Review by kosara-g — DISMISSED** [review]

**Review by timjon — CHANGES_REQUESTED** [review]
compile.sh version number

**Review by timjon — DISMISSED** [review]

**Review by Sebastian-Fransson — DISMISSED** [review]
Gut

**Review by kosara-g — DISMISSED** [review]
Still good.

**Review by Sebastian-Fransson — APPROVED** [review]
Again

**Review by timjon — APPROVED** [review]
